uxarray/grid/neighbors.py

```diff
@@ -0,0 +1,220 @@
+
+        # construct tree from corner nodes
+        if node_type == "corner":
+
+            XY = np.vstack((deg2rad(grid.Mesh2_node_y.values),
+                            deg2rad(grid.Mesh2_node_x.values))).T
+
+            self.n_elements = grid.nMesh2_node
+            self.tree = SKBallTree(XY, metric='haversine')
+
+        # construct tree from center nodes
+        elif node_type == "center":
+
+            # warning until we work on how face centers are represented and constructed
+            warnings.warn(
+                "Internal representation of face center coordinates has not been tested or verified. Results"
+                "may not be as expected until this has been patched in the Grid class."
+            )
+
+            # center nodes must be present, remove once construction methods for them is added
+            if grid.Mesh2_face_x is None or grid.Mesh2_face_y is None:
+                raise ValueError
+
+            XY = np.vstack((deg2rad(grid.Mesh2_face_y.values),
+                            deg2rad(grid.Mesh2_face_x.values))).T
+
+            self.n_elements = grid.nMesh2_face
+            self.tree = SKBallTree(XY, metric='haversine')
+
+    def query(self,
+              xy: Union[np.ndarray, list, tuple],
+              k: Optional[int] = 1,
+              use_radians: Optional[bool] = False,
+              return_distance: Optional[bool] = True,
+              dualtree: Optional[bool] = False,
+              breadth_first: Optional[bool] = False,
+              sort_results: Optional[bool] = True):
+        """Query the tree for the ``k`` nearest neighbors.
+
+        Parameters
+        ----------
+        xy : array_like
+            coordinate pairs in degrees (lon, lat) to query
+        k: int, default=1
+            The number of nearest neighbors to return
+        use_radians : bool, optional
+            if True, queries assuming xy are inputted in radians, not degrees
+        return_distance : bool, optional
+            if True, return a tuple ``(d, i)`` of distances and indices if False, return array i
+        dualtree : bool, default=False
+            if True, use the dual tree formalism for the query: a tree is built for the query points, and the pair of trees is used to efficiently search this space. This can lead to better performance as the number of points grows large.
+        breadth_first : bool, default=False
+            if True, then query the nodes in a breadth-first manner. Otherwise, query the nodes in a depth-first manner.
+        sort_results : bool, default=True
+            if True, then distances and indices of each point are sorted on return, so that the first column contains the closest points. Otherwise, neighbors are returned in an arbitrary order.
+
+        Returns
+        -------
+        d : ndarray of shape (xy.shape[0], k), dtype=double
+            Each entry gives the list of distances to the neighbors of the corresponding point.
```

Comment:
A bit difficult to read this. Did you mean something like "Distance array that keeps the distances of the k-nearest neighbors to the entries from `xy` in each row"